snapshots/store.go

```diff
@@ -91,7 +91,7 @@ func (s *Store) Get(height uint64, format uint32) (*types.Snapshot, error) {
 
 // Get fetches the latest snapshot from the database, if any.
 func (s *Store) GetLatest() (*types.Snapshot, error) {
-	iter, err := s.db.ReverseIterator(encodeKey(0, 0), encodeKey(math.MaxUint64, math.MaxUint32))
+	iter, err := s.db.ReverseIterator(encodeKey(0, 0), encodeKey(uint64(math.MaxUint64), math.MaxUint32))
```

Comment:
Why this is needed? Isn't `math.MaxUint64` already uint64?

Reply:
All casts are related to [this Go specification](https://github.com/golang/go/issues/23086)

Reply:
No, it's an untyped constant: https://golang.org/ref/spec#Constants (not a bug, a feature! :))

Reply:
Yes, exactly. constants will adapt to the expected types. The `encodeKey` has following declaration:
```
func encodeKey(height uint64, format uint32) []byte
```
so the `math.MaxUint64` will automatically become `uint64` - seems that this conversion is not needed.

Reply:
I was actually re-reviewing this and you're right. In `master` we should revise the code patch as follows:

1. revert it
2. build multi-arch
3. fix build errors one by one